pkg/server/registry/apigroups/acorn/apps/validator.go

```diff
@@ -171,35 +171,37 @@ func (s *Validator) Validate(ctx context.Context, obj runtime.Object) (result fi
 			return
 		}
 
-		disableCheckImageAllowRules := false
-		if app.Spec.Stop != nil && *app.Spec.Stop {
-			// app was stopped, so we don't need to check image allow rules (this could prevent stopping an app if the image allow rules changed)
-			disableCheckImageAllowRules = true
-		}
-
-		if !disableCheckImageAllowRules {
+		if !z.Dereference(app.Spec.Stop) {
```

Comment:
For the stop case, is this good enough? or do we want to validate that _nothing else_ has changed but the `spec.stop` field? I'm worried about the case where other fields are changed on stop; e.g. I switch to a bad compute class.

Reply:
So with your code, we stop an App and skip validation for imageAllowRules (previous behavior) and computeClasses (new). If I swap to a bad ComputeClass at the same time nothing will be validated. If I then start the App again it will go through validation for imageAllowRules and computeClasses and fail the computeClass validation. 

That behavior could be fine but could also be confusing if a user stopped an App a long time ago and starts an App up again to be met with a "invalid compute class" error. However, the behavior was the same for imageAllowRules before this PR, so its likely fine.

Reply:
>  However, the behavior was the same for imageAllowRules before this PR, so its likely fine

yeah, this was my sentiment exactly

Reply:
I'm going to consider this resolved for now.